src/FEValues/face_integrals.jl

```diff
@@ -149,3 +94,34 @@ function weighted_normal(J::Tensor{2,3}, ::Type{RefTetrahedron}, face::Int)
     end
     throw(ArgumentError("unknown face number: $face"))
 end
+
+###################
+# All RefPrism 3D #
+###################
+
+function weighted_normal(J::Tensor{2,3}, ::Type{RefPrism}, face::Int)
+    @inbounds begin
+        face == 1 && return J[:,2] × J[:,1]
+        face == 2 && return J[:,1] × J[:,3]
+        face == 3 && return J[:,3] × J[:,2]
+        face == 4 && return (J[:,2]-J[:,1]) × J[:,3]
+        face == 5 && return J[:,1] × J[:,2]
+    end
+    throw(ArgumentError("unknown face number: $face"))
+end
+
+#####################
+# All RefPyramid 3D #
+#####################
+
+function weighted_normal(J::Tensor{2,3}, ::Type{RefPyramid}, face::Int)
+    @inbounds begin
+        face == 1 && return J[:,2] × J[:,1]
+        face == 2 && return J[:,1] × J[:,3]
+        face == 3 && return J[:,3] × J[:,2]
+        face == 4 && return J[:,2] × (J[:,3]-J[:,1])
+        face == 5 && return (J[:,3]-J[:,2]) × J[:,1]
+    end
+    throw(ArgumentError("unknown face number: $face"))
```

Comment:
btw it looks like these throws are redundant because there are outer checks IIUC. Should I remove them or add tests for them anyways? (IDK if user would access face weighted normals directly through this function instead of using facevalues)

Reply:
If you can show that there is a significant performance benefit by removing the error (could be), then that would be fine, but should probably be called `_unsafe_weighted_normal` or something (I don't think we do this at other places in Ferrite at the moment, although `shape_value(::Interpolation, ...)` could be a candidate, xref #766 )

But I would leave that for a separate PR

Reply:
I don't see performance benefit (or maybe I should use something other than `@benchmark` to benchmark operations that take about 1ns :"D)

Maybe I can add some tests for these error paths then to improve test coverage.

Reply:
Yes, it is hard to benchmark due to const propagation I can imagine. But let's leave that for a separate issue.